### ISSUE TITLE
IDE0004 false positive in casting list of nullable reference elements to non-nullable

### ISSUE THREAD

**Issue by rhys-vdw** [dotnet/roslyn#83284]
**Version Used**: 
Sorry not sure how to check, but it's VSCode: `ms-dotnettools.csharp` 2.130.5

**Steps to Reproduce**:

Repro:
```cs
class Example {
  static List<object> Case() {
    var lines = new List<object?>();
    return new(lines as List<object>);
  }
}
```

<img width="706" height="231" alt="Image" src="https://github.com/user-attachments/assets/ef4e9003-a776-4004-abee-c9e29440c667" />

<img width="1251" height="279" alt="Image" src="https://github.com/user-attachments/assets/6c369f4b-1b5c-4510-9c24-6d8477ef46ed" />

**Diagnostic Id**: [IDE0004: Cast is redundant.](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0004)

**Expected Behavior**:
Cast is not flagged as redundant.

**Actual Behavior**:
Cast is flagged as redundant, despite resolving [CS8620](https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS8620)).

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/49690

**Comment by rhys-vdw** [comment]
Does appear to be related to #49690, but that was closed 5 years ago. Perhaps closed in error.
